### PR TITLE
feat(catalog): IngestBangumi pipeline through ports (CATALOG-7 #948)

### LINKED ISSUE #948: CATALOG-7 — Own the complete Bangumi ingest lifecycle

### PR DESCRIPTION
**Pull request by lifeodyssey**
Closes #948 (parent #936). Wave 2.

- IngestBangumi owns claim→fetch→raw→enrich→publish→completion + negative-cache outcome through source/store/publisher ports; on-demand + scheduled callers migrated.
- Deleted work vocabulary, orchestrator, adapter-hidden orchestration, duplicate state machines, handler retry policy.
- Red tests: singleflight, negative cache TTL, retryable upstream, atomic publish, crash recovery, idempotent replay (10 unit + 7 spike).
- Review fixes: class split ≤50 lines, test files ≤200, work-points migrated onto IngestLifecycle (state machine once), stale comments, error root-cause preserved.
- Two-axis review: APPROVE. Gates: typecheck, oxlint, 395 tests.

## Summary by Sourcery

Introduce a dedicated IngestBangumi use case that owns the full Bangumi ingest lifecycle behind source/store/publisher ports, and migrate API, cron, and entrypoint callers from the old orchestrator to this new pipeline.

New Features:
- Add the IngestBangumi pipeline composed over source, store, and publisher ports with configurable negative-cache TTLs for failures and empty results.

Enhancements:
- Refactor work-points, search, and internal ingest entrypoint to depend on the new IngestBangumi lifecycle interface instead of bespoke claim/guard logic.
- Rename seed and cron ingest utilities from generic work-centric naming to Bangumi-focused terminology for clarity.
- Align enrich, publish, raw-store, and preview layers on Bangumi-centric identifiers and shared ingest types.

Tests:
- Add unit and spike tests covering IngestBangumi singleflight behavior, negative caching, crash recovery, idempotent replay, and atomic publish ordering.
- Update existing tests for work-points, cron handlers, preview, search, and ingest entrypoint to target the new IngestBangumi interfaces and Bangumi-focused naming.

### CODE REVIEW

File: workers/catalog/src/ingest/ingest-bangumi.ts [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+}
+
+/** Negative-cache the failure (clears the 'running' row) and report it. */
+async function fail(runtime: IngestRuntime, bangumiId: string, errorCode: string, reason: string): Promise<IngestResult> {
+  const ttlSeconds = errorCode === IngestErrorCode.NotFound
+    ? runtime.ttl.emptySeconds
+    : runtime.ttl.failureSeconds;
+  await runtime.store.markFailed(bangumiId, { errorCode, ttlSeconds, error: reason });
+  const status = errorCode === IngestErrorCode.NotFound ? "empty" : "failed";
+  return { status, reason };
```
[the remark this answers is not in the document]
Comment (lifeodyssey) [reply workers/catalog/src/ingest/ingest-bangumi.ts]: fixed: fail(bangumiId, errorCode) tightened to the IngestErrorCode union so TTL/error coupling is typed — review-fix commit.
